Sources/LogInOut.php

```diff
@@ -60,6 +60,9 @@ function Login()
 	// Set the login URL - will be used when the login process is done (but careful not to send us to an attachment).
 	if (isset($_SESSION['old_url']) && strpos($_SESSION['old_url'], 'dlattach') === false && preg_match('~(board|topic)[=,]~', $_SESSION['old_url']) != 0)
 		$_SESSION['login_url'] = $_SESSION['old_url'];
+	// This came from a valid hashed return url.  Or something that knows our secrets...
+	elseif (!empty($_REQUEST['return_hash']) && !empty($_REQUEST['return_to']) && md5($_REQUEST['return_to'] . $image_proxy_secret) == $_REQUEST['return_hash'])
```

Comment:
Please use `hash_hmac('sha1', $_REQUEST['return_to'], $image_proxy_secret)` instead of `md5($_REQUEST['return_to'] . $image_proxy_secret)`. This is important if we want to keep the secret actually secret.